### PR TITLE
Link to search modal

### PR DESCRIPTION
**Pull request by dojutsu-user**
Demo:

If the url params contains `rtd_search`, then the modal will open and search for its value.

Peek 2019-07-01 12-58

The demo is not according to the PR #19 
But I tried it with #19 and it worked great.

### CODE REVIEW

File: sphinx_search/_static/js/rtd_sphinx_search.js [hunk 1]
```diff
@@ -521,12 +588,66 @@ window.addEventListener("DOMContentLoaded", evt => {
...
+                SEARCH_QUERY !== null &&
+                SEARCH_QUERY !== ""
+            ) {
+                let urlParam = convertObjToUrlParams({
+                    rtd_search: encodeURIComponent(SEARCH_QUERY)
+                });
+                urlToBeCopied += "?" + urlParam;
+            }
+
+            copyToClipBoard(urlToBeCopied);
```
Comment (ericholscher) [inline sphinx_search/_static/js/rtd_sphinx_search.js]: I don't quite follow what this is doing. We should just update the browser's URL when the user types with the proper data, we don't need a custom way to copy a link.

File: sphinx_search/_static/js/rtd_sphinx_search.js [hunk 2]
```diff
@@ -406,19 +418,74 @@ const removeSearchModal = () => {
...
+ */
+const getUrlVars = () => {
+    let vars = {};
+    let params = window.location.href.replace(
+        /[?&]+([^=&]+)=([^&]*)/gi,
+        (m, key, value) => {
+            vars[key] = value;
+        }
+    );
+    return vars;
```
Comment (ericholscher) [inline sphinx_search/_static/js/rtd_sphinx_search.js]: Is this really not in jQuery already? I feel like we're reinventing a lot of wheels.

File: sphinx_search/_static/js/rtd_sphinx_search.js [hunk 3]
```diff
@@ -406,19 +418,74 @@ const removeSearchModal = () => {
...
+    let textarea = createDomNode("textarea", {
+        readonly: "",
+        style: "position: absolute; left: -9999px;"
+    });
+    textarea.value = str;
+    document.body.appendChild(textarea);
+    textarea.select();
+    document.execCommand("copy");
+    document.body.removeChild(textarea);
+};
```
Comment (ericholscher) [inline sphinx_search/_static/js/rtd_sphinx_search.js]: Same here, is this really not a standard thing?

File: sphinx_search/_static/js/rtd_sphinx_search.js (hunk 2 above)
Comment (dojutsu-user) [reply sphinx_search/_static/js/rtd_sphinx_search.js]: No.
I checked on the internet. There is no direct utility from the jquery or javascript to get the url parameters.

File: sphinx_search/_static/js/rtd_sphinx_search.js (hunk 3 above)
Comment (dojutsu-user) [reply sphinx_search/_static/js/rtd_sphinx_search.js]: No.
here also -- no utility of jquery or javascript is there which dirctly copies the text.

File: sphinx_search/_static/js/rtd_sphinx_search.js (hunk 1 above)
Comment (dojutsu-user) [reply sphinx_search/_static/js/rtd_sphinx_search.js]: This is generating the url to the search ui and then copying it to the clipboard.

**Comment by dojutsu-user** [comment]
> I don't think this is the right approach for sharing. We should simply update the rtd_search query param as the user types, and then they can just share the URL as they would normally.

I have some different thoughts here -- I think it would be better to give user the choice to share the link to his/her search. Just like the sections in the docs.
What are your thoughts on this? @ericholscher

**Comment by dojutsu-user** [comment]
Closing in favor of #25

**Pull request closed without merging**